docs/docsite/rst/testing_modules.rst

```diff
@@ -0,0 +1,196 @@
+  + After the operations, the test verifies the output files and asserts that the changes occurred as expected or not changed when re-run in check_mode, confirming the module's correct behavior and idempotency.  
+
+This structured approach ensures thorough testing of the modules' functionality and stability in different modes and repeated executions.
+
+
+5. We begin by adding tasks to post configuration to the APIC. This includes creation of all the classes such as tenant and l3out that were used in the construct_url function in our module.
+
+.. code-block:: yaml
+
+      - name: Add a new tenant
+        aci_tenant:
+          <<: *aci_info 
+          tenant: ansible_tenant
+          description: Ansible tenant
+          state: present
+
+      - name: Add a new parent object
+        aci_l3out:
+          <<: *aci_info
+          tenant: ansible_tenant
+          name: '<parent_object_name>' # Replace <parent_object_name> with the actual name of the parent object, e.g., ansible_l3out
+          description: Parent object description
+          state: present
+
+
+.. code-block:: text
+
+6. The next section consists of adding tasks for all aspects of our module. We include Ansible's register attribute to save the result of the task. The procedure is as follows:
+
+* Tasks to clean the environment to begin with a clean slate.
+        * Task to add a new object with only the required parameters. (check mode) -> where the check_mode is set to true.
+        * Task to add a new object with only the required parameters. (actual run) -> where the check_mode is set to false.
+        * task to add a new object with only the required parameters again. (idempotency), to test the idempotency of the module.
+        * An assert task to verify the object was created successfully for the previous tasks of adding new object with only the required parameters.(check mode, actual run and idempotency).
+        * Task to add a new object with all the parameters. (check mode) -> where the check_mode is set to true.
+        * Task to add a new object with all the parameters. (actual run) -> where the check_mode is set to false.
+        * Task to add a new object with all the parameters again. (actual run), to test the idempotency of the module.
+        * An assert task to verify the object was created successfully for the previous tasks of adding new object with all the parameters. (check mode, actual run and idempotency).
+        * Task to update the object parameters. (check mode) -> where the check_mode is set to true.
+        * Task to update the object parameters. (actual run) -> where the check_mode is set to false.
+        * Task to update the object parameters again. (actual run), to test the idempotency of the module.
+        * An assert task to verify the object was updated successfully for the previous tasks of updating object parameters. (check mode, actual run and idempotency).
```

Comment:
are we not combining these steps in our testing? so we update with full confgiuration